sdk/data/aztables/README.md

```diff
@@ -1,151 +1,342 @@
+
+```golang
+cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+handle(err)
+service, err := aztables.NewServiceClient("https://<myAccountName>.table.core.windows.net", cred, nil)
+
+resources := aztables.AccountSASResourceTypes{Service: true}
+permission := aztables.AccountSASPermissions{Read: true}
+start := time.Date(2021, time.August, 21, 1, 1, 0, 0, time.UTC)
+expiry := time.Date(2022, time.August, 21, 1, 1, 0, 0, time.UTC)
+sasUrl, err := service.GetAccountSASToken(resources, permission, start, expiry)
+handle(err)
+
+sasService, err := aztables.NewServiceClient(sasUrl, azcore.AnonymousCredential(), nil)
+handle(err)
+```
 
-#### Creating a Cosmos DB
-
-Create a Cosmos DB account `MyCosmosDBDatabaseAccount` in resource group `MyResourceGroup`
-in the subscription `MySubscription` and a table named `MyTableName` in the account.
-
-
-### Authenticate the Client
-
-Learn more about options for authentication _(including Connection Strings, Shared Key, and Shared Key Signatures)_ in our samples.
 
 ## Key concepts
+Common uses of the table service include:
+* Storing TBs of structured data capable of serving web scale applications
+* Storing datasets that do not require complex joins, foreign keys, or stored procedures and can be de-normalized for fast access
+* Quickly querying data using a clustered index
+* Accessing data using the OData protocol filter expressions
+
+The following components make up the Azure Tables Service:
+* The account
+* A table within the account, which contains a set of entities
+* An entity within a table, as a dictionary
+
+The Azure Tables client library for Go allows you to interact with each of these components through the
+use of a dedicated client object.
+
+### Clients
+Two different clients are provided to interact with the various components of the Table Service:
+1. **`ServiceClient`** -
+    * Get and set account settings
+    * Query, create, and delete tables within the account.
+    * Get a `Client` to access a specific table using the `NewClient` method.
+2. **`Client`** -
+    * Interacts with a specific table (which need not exist yet).
+    * Create, delete, query, and upsert entities within the specified table.
+    * Create or delete the specified table itself.
+
+### Entities
+Entities are similar to rows. An entity has a **`PartitionKey`**, a **`RowKey`**, and a set of properties. A property is a name value pair, similar to a column. Every entity in a table does not need to have the same properties. Entities are returned as JSON, allowing developers to use JSON marshalling and unmarshalling techniques. Additionally, you can use the `aztables.EDMEntity` to ensure proper round-trip serialization of all properties.
```

Comment:
Might be a chicken-and-egg thing but can you link to the godoc for EDMEntity in this paragraph?

Reply:
No I can't do this because it isn't published yet. After the preview I can update and include the link